### ISSUE TITLE
curriculum learning is not working

### ISSUE THREAD

**Issue by jwson97** [Unity-Technologies/ml-agents#4465]
I wrote the yaml file as below.
   
```
behaviors:
    AircraftPlayer:
        trainer_type: ppo
        hyperparameters:
            batch_size : 2048
            beta: 1.0e-2
            buffer_size: 20480
            epsilon: 0.2
            lambd: 0.95
            learning_rate: 3.0e-4
            learning_rate_schedule: linear
            num_epoch: 3
        network_settings:
            hidden_units: 256
            memory:
                memory_size : 256
                sequence_length: 64
            normalize : false
            num_layers: 2
            vis_encoder_type: simple
        reward_signals:
            extrinsic:
                strength: 1.0
                gamma: 0.99
        max_steps: 5.0e6
        time_horizon: 128
        summary_freq: 32000
        threaded: true

environment_parameters:
    checkpoint_radius:
        curriculum :
          - name : First
            completion_criteria:
                measure : reward
                behavior : AircraftPlayer
                signal_smoothing : true
                min_lesson_length : 200
                threshold : 2.0
            value : 70.0
          - name : Second
            completion_criteria:
                measure : reward
                behavior : AircraftPlayer
                signal_smoothing : true
                min_lesson_length : 200
                threshold : 2.0
            value : 50.0
          - name : Third
            completion_criteria:
                measure : reward
                behavior : AircraftPlayer
                signal_smoothing : true
                min_lesson_length : 200
                threshold : 2.0
            value : 30.0
```

And the code in the script is as follows.

`                
if (localCheckpointDir.magnitude < Academy.Instance.EnvironmentParameters.GetWithDefault("checkpoint_radius", 30.0f))
{ 
    GotCheckpoint();
}
`

However, after training, curriculum learning does not work. And even more strangely, if I look at the value in "Lesson/checkpoint_radius" in tensorboard, even the default value in the current code is set to 30.0f, which is displayed as 0. What's the problem?

run command like this:
`mlagents-learn config/AircraftPlayer.yaml --run-id=AircraftPlayer_curriculum`

![Screen Shot 2020-09-09 at 12 11 12 PM](https://user-images.githubusercontent.com/42993434/92550028-9e69bc80-f295-11ea-9b02-d1960fc6d0f9.png)

**Comment by hvpeteet** [comment]
Hi @jwson97,

To be able to better understand what is going wrong would you be able to share the following graphs with us from Tensorboard?

1. Environment -> Cumulative Reward
1. Losses -> Value Loss
1. Policy -> Entropy

**Comment by jwson97** [comment]
The graph examined in tensorboard is as follows.
![Screen Shot 2020-09-14 at 9 53 11 AM](https://user-images.githubusercontent.com/42993434/93033222-486b8f00-f670-11ea-8f13-6c8d043971ba.png)
![Screen Shot 2020-09-14 at 9 53 19 AM](https://user-images.githubusercontent.com/42993434/93033224-4c97ac80-f670-11ea-82cb-ea474076d6fa.png)

**Comment by hvpeteet** [comment]
We tried to reproduce a similar error on our example WallJump environment (also uses curriculum learning) but it seemed to work correctly. Since we can't get a reasonable reproduction could you try running with `progress  instead of `reward` and set the `threshold` fields to a 0.0-1.0 value indicating a fraction of `max_steps`. We are trying to figure out if this may be an issue with using `reward`. Thanks

**Comment by jwson97** [comment]
Thanks for your efforts.

**Comment by hvpeteet** [comment]
Always happy to help, did you find a solution or did the issue persist when using progress?

**Comment by AcelisWeaven** [comment]
Hi, just wanted to say I've been having the same issue both on a custom env (with `progress`) and on WallJump_curriculum.

Here's my setup :
- **ml-agents**: releases 8 and 9
- **Unity**: 2018.4/2020.2 (both trained in the Editor directly)
- **OS**: Windows
- **Python**: Python 3.7.9

(Even using a low `threshold` with `progress` didn't help)
